src/equals.ts

```diff
@@ -0,0 +1,73 @@
+type UnknownProperties = {
+  [property: string]: unknown
+}
+
+// compare compares two values that can be any type additionally it takes an optional property which has the ignored fields.
+export function equals<T>(a: T, b: T, ignoredFields?: string[]): boolean {
+  return internalEquals(a, b, '', ignoredFields)
+}
+
+function internalEquals<T>(a: T, b: T, parentPath: string, ignoredFields?: string[], isPartial = false): boolean {
+  if (Array.isArray(a)) {
+    return arrayEquals(a, b as unknown[], ignoredFields)
+  } else if (typeof a === 'object' && typeof b === 'object') {
+    return internalObjectEquals(a as UnknownProperties, b as UnknownProperties, parentPath, ignoredFields)
+  }
+
+  return a === b || (isPartial && !!a && !b)
+}
+
+function arrayEquals<T extends unknown[]>(a: T, b: T, ignoredFields?: string[]): boolean {
+  if (a.length !== b.length) {
+    return false
+  }
+
+  for (let i = 0; i < a.length; i++) {
+    if (!internalEquals(a[i], b[i], '', ignoredFields)) {
+      return false
+    }
+  }
+
+  return true
+}
+
+function internalObjectEquals<T extends UnknownProperties>(
+  a: T,
+  b: T,
+  parentPath: string,
+  ignoredFields?: (keyof T)[],
+  isPartial = false,
+): boolean {
+  const keysA = Object.keys(a)
+  const keysB = Object.keys(b)
+  if (keysA.length !== keysB.length) {
+    return false
+  }
+
+  for (const key of keysA) {
+    const path = parentPath ? `${parentPath}.${key}` : key
+
+    if (ignoredFields && ignoredFields.indexOf(path) !== -1) {
+      continue
+    }
+
+    const valueA = a[key]
+    const valueB = b[key]
+
+    const equals = internalEquals(valueA, valueB, path, ignoredFields as string[], isPartial)
+
+    if (!equals) {
+      return false
+    }
+  }
+  return true
+}
+
+export function objectEquals<T extends UnknownProperties>(
+  a: T,
+  b: T,
+  ignoredFields?: (keyof T)[],
+  isPartial = false,
+): boolean {
+  return internalObjectEquals(a, b, '', ignoredFields, isPartial)
+}
```

Comment:
Strongly opposed to adding another file to the source tree.

Reply:
Removed the file